pkg/stub/psmdb.go

```diff
@@ -160,8 +156,39 @@ func (h *Handler) newStatefulSetContainers(m *v1alpha1.PerconaServerMongoDB, rep
 	return containers
 }
 
+// newStatefulSetVolumes returns a slice of volumes PSMDB stateful set
+func (h *Handler) newStatefulSetVolumes(m *v1alpha1.PerconaServerMongoDB) ([]corev1.Volume, error) {
+	volumes := []corev1.Volume{
+		{
+			Name: m.Spec.Secrets.Key,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					DefaultMode: &secretFileMode,
+					SecretName:  m.Spec.Secrets.Key,
+					Optional:    &util.FalseVar,
+				},
+			},
+		},
+	}
+
+	if h.hasBackupsEnabled(m) {
+		agentVolumes, err := h.backups.NewAgentVolumes()
+		if err != nil {
+			return nil, err
+		}
+		volumes = append(volumes, agentVolumes...)
+	}
```

Comment:
It's not a merge-blocker just a comment on design:
I believe better move the NewAgentVolumes() out of this method. Now I see few disadvantages:
1) Less flexibility. Let say in case NewAgentVolumes() returns error it will fail all execution. May be that's a desired behavior but better to give the opportunity to decide on the caller level.
2) Application logic become more complex and tricky since we have many different checks on different levels for the same thing. So it's miss this pitfalls.

Reply:
Good point!

As .NewAgentVolumes() logs any errors using logrus.Errorf() I think it doesn't make sense to throw an error if .NewAgentVolumes() fails.

I have modified .newStatefulSetVolumes() to return []corev1.Volume only with no error so the statefulset can still function if there is a problem with creating the backup-agent config.